### PR TITLE
Workflow is required always, even when no default workflow

### PR DESCRIPTION
**Pull request by fiskus**
Workflow can be optional for package creation. But UI element is mandatory and should have value always

Screenshot from 2020-11-24 12-37-18

### CODE REVIEW

**Comment by fiskus** [comment]
@nl0 **Why `isRequired` removed**
UI data is not identical to config.yml. Config doesn't have "No workflow" option, UI has. UI always have some workflow - some workflow from config.yml or "No workflow". Thus UI always required workflow, and this property is not needed.

File: catalog/app/containers/Bucket/PackageDialog/SelectWorkflow.js [hunk 1]
```diff
@@ -51,6 +62,7 @@ export default function SelectWorkflow({ className, disabled, items, onChange, v
         ))}
       </M.Select>
       <M.FormHelperText>
+        {error && <span className={classes.error}>{error}</span>}
```
Comment (nl0) [inline catalog/app/containers/Bucket/PackageDialog/SelectWorkflow.js:65]:
```suggestion
        {!!error && <span className={classes.error}>{error}</span>}
```

File: catalog/app/containers/Bucket/PackageDialog/PackageDialog.js [hunk 2]
```diff
@@ -190,6 +191,7 @@ export function WorkflowInput({ input, meta, workflowsConfig }) {
       onChange={input.onChange}
       value={input.value}
       disabled={disabled}
+      error={errors[errorKey]}
```
Comment (nl0) [inline catalog/app/containers/Bucket/PackageDialog/PackageDialog.js:194]:
```suggestion
      error={errorKey ? (errors[errorKey] || errorKey) : undefined}
```

File: catalog/app/containers/Bucket/PackageDialog/PackageDialog.js [hunk 3]
```diff
@@ -178,10 +178,11 @@ const useWorkflowInputStyles = M.makeStyles((t) => ({
   },
 }))
 
-export function WorkflowInput({ input, meta, workflowsConfig }) {
+export function WorkflowInput({ errors, input, meta, workflowsConfig }) {
```
Comment (nl0) [inline catalog/app/containers/Bucket/PackageDialog/PackageDialog.js:181]:
```suggestion
export function WorkflowInput({ input, meta, workflowsConfig, errors = {} }) {
```

**Comment by nl0** [comment]
> **Why `isRequired` removed**
> UI data is not identical to config.yml. Config doesn't have "No workflow" option, UI has. UI always has some workflow - some workflow from config.yml or "No workflow". Thus UI always requires workflow, and this property is not needed.

ok that makes sense

**Review by nl0 — APPROVED** [review]

File: catalog/app/containers/Bucket/PackageDialog/PackageDialog.js (hunk 2 above)
Comment (fiskus) [reply catalog/app/containers/Bucket/PackageDialog/PackageDialog.js:194]: `errors[errorKey] || errorKey` seems enough, as `{}[undefined]` will be `undefined`

Comment (fiskus) [reply catalog/app/containers/Bucket/PackageDialog/PackageDialog.js:194]: But this works for me as well

Comment (nl0) [reply catalog/app/containers/Bucket/PackageDialog/PackageDialog.js]: > `{}[undefined]` will be `undefined`

well, in theory there may be something under `"undefined"` key (`{ undefined: 1 }[undefined] === 1`)

**Pull request merged**